src/main/java/graphql/parser/Parser.java

```diff
@@ -238,21 +238,42 @@ public void syntaxError(Recognizer<?, ?> recognizer, Object offendingSymbol, int
     }
 
     private void setupParserListener(MultiSourceReader multiSourceReader, GraphqlParser parser, GraphqlAntlrToLanguage toLanguage) {
-        int maxTokens = toLanguage.getParserOptions().getMaxTokens();
+        ParserOptions parserOptions = toLanguage.getParserOptions();
+        ParsingListener parsingListener = parserOptions.getParsingListener();
+        int maxTokens = parserOptions.getMaxTokens();
         // prevent a billion laugh attacks by restricting how many tokens we allow
         ParseTreeListener listener = new GraphqlBaseListener() {
             int count = 0;
 
             @Override
             public void visitTerminal(TerminalNode node) {
+
+                final Token symbol = node.getSymbol();
```

Comment:
This probably cant be null.....but go we want a NPE or not?  I vote for safety over optimised correctness.

ANTLR is a complicated beast - will it always be non null?